File: http/http-client-annotation-processor/src/test/java/ru/tinkoff/kora/http/client/annotation/processor/client/ClientWithQueryParams.java
package ru.tinkoff.kora.http.client.annotation.processor.client;


import ru.tinkoff.kora.http.client.common.annotation.HttpClient;
import ru.tinkoff.kora.http.common.HttpMethod;
import ru.tinkoff.kora.http.common.annotation.Header;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.http.common.annotation.Query;

import javax.annotation.Nullable;
import java.time.LocalDate;
import java.util.List;

@HttpClient(configPath = "clientWithQueryParams")
public interface ClientWithQueryParams {
    @HttpRoute(method = HttpMethod.POST, path = "/test1?test=test")
    void test1(@Query String test1);

    @HttpRoute(method = HttpMethod.POST, path = "/test2?")
    void test2(@Query("test2") String test);

    @HttpRoute(method = HttpMethod.POST, path = "/test3")
    void test3(@Query("test3") String test);

    @HttpRoute(method = HttpMethod.POST, path = "/test4")
    void test4(@Query("test4") String test4, @Nullable @Query("test") String test);

    @HttpRoute(method = HttpMethod.POST, path = "/test5")
    void test5(@Query String test51, @Query String test52, @Query String test53, @Query String test54, @Nullable @Query String test55, @Nullable @Query String test56);

    @HttpRoute(method = HttpMethod.POST, path = "/test6")
    void test6(@Nullable @Query String test61, @Nullable @Query String test62, @Nullable @Query String test63);

    @HttpRoute(method = HttpMethod.POST, path = "/nonStringParams")
    void nonStringParams(@Query int query1, @Query Integer query2);

    @HttpRoute(method = HttpMethod.POST, path = "/multipleParams")
    void multipleParams(@Query List<String> query1, @Nullable @Query List<Integer> query2);


}

File: http/http-client-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/client/symbol/processor/client/ClientWithQueryParams.kt
package ru.tinkoff.kora.http.client.symbol.processor.client

import ru.tinkoff.kora.http.client.common.annotation.HttpClient
import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.http.common.annotation.Query
import java.time.LocalDate

@HttpClient(configPath = "clientWithQueryParams")
interface ClientWithQueryParams {
    @HttpRoute(method = HttpMethod.POST, path = "/test1?test=test")
    fun test1(@Query test1: String?)

    @HttpRoute(method = HttpMethod.POST, path = "/test2?")
    fun test2(@Query("test2") test: String)

    @HttpRoute(method = HttpMethod.POST, path = "/test3")
    fun test3(@Query("test3") test: String)

    @HttpRoute(method = HttpMethod.POST, path = "/test4")
    fun test4(@Query("test4") test4: String?, @Query("test") test: String?)

    @HttpRoute(method = HttpMethod.POST, path = "/test5")
    fun test5(
        @Query test51: String?,
        @Query test52: String?,
        @Query test53: String?,
        @Query test54: String?,
        @Query test55: String?,
        @Query test56: String?
    )

    @HttpRoute(method = HttpMethod.POST, path = "/test6")
    fun test6(@Query test61: String?, @Query test62: String?, @Query test63: String?)

    @HttpRoute(method = HttpMethod.POST, path = "/nonStringParams")
    fun nonStringParams(@Query query1: Int, @Query query2: LocalDate, @Query query3: Int?)

    @HttpRoute(method = HttpMethod.POST, path = "/multipleParams")
    fun multipleParams(@Query query1: List<Int>, @Query query2: Iterable<String>?, @Query query3: List<LocalDate?>)
}
